libs/protocol-client/src/lib/client/contribution.ts

```diff
@@ -180,6 +243,70 @@ export class Contribution extends BaseClient {
       },
     });
   }
+
+  private async _updateUserOnChainContribution(contribution: {
+    id: number;
+    activityTypeId: number;
+    userId: number;
+    args: MintArgs;
+    name: Uint8Array;
+    details: Uint8Array;
+    proof: Uint8Array;
+    txHash: string;
+    onChainId: number;
+  }) {
+    console.log('id in the update:', contribution.id);
+    const updateResponse = this.sdk.updateUserOnChainContribution({
+      data: {
+        name: ethers.utils.toUtf8String(contribution.name),
```

Comment:
To make these a little less verbose you can destructure the keys of contribution
`const {args, name, details, proof, onChainId, userId, id, txHash} = contribution`